### PR TITLE
To be independent of keyboard layout; replaced VK codes with ASCII chars.

### PR DESCRIPTION
**Pull request by visorz**
Hello, this is my first pull request ever. I changed the Main.cpp to compare the typed ASCII symbols with given symbols. That works over different languages and keyboard layouts. All code I added uses self-explaining variable names, named state comparisons and extracted methods. Overall the code is shorter and easier to extend.

### CODE REVIEW

**Comment by dail8859** [comment]
Hi and thanks for the PR! Does this in theory do the same thing as PR #5?

**Comment by visorz** [comment]
PR #5 uses the same mechanism: translating the input key to an ASCII character and then comparing. I had the idea independently of #5 and it was a conincidence, that we both worked on the same layout independence. My PR #6 includes all braces that have been added in PR #5 but I did not update the `Readme` file in my fork. That is why I did not include it in the commit.

My code does not include `goto` and does not use `if .. else ..` statements for all possible combinations. Therefore I am convinced, that my code is easier to read and to maintain. Please compile it and try it yourself.

File: src/Main.cpp [hunk 1]
```diff
@@ -116,49 +117,64 @@ static void SurroundSelectionsWith(char ch1, char ch2) {
...
+
+	int isDeadKey = ToAscii(
+		virtualKeyCode,
+		scanCode,
+		keyboardState,
+		&charResult,
+		isMenuActive
+	);
+
+	return (CHAR)charResult;
```
Comment (MAPJe71) [inline src/Main.cpp]: C++ style type-casting is preferred.
Cast to `char` as that is the return-type of the function.

File: src/Main.cpp [hunk 2]
```diff
@@ -116,49 +117,64 @@ static void SurroundSelectionsWith(char ch1, char ch2) {
...
+	GetKeyboardState(keyboardState);
+
+	// https://msdn.microsoft.com/en-us/library/ms644984(v=VS.85).aspx
+	UINT scanCode = HIBYTE(LOWORD(lparam)); 
+	UINT virtualKeyCode = UINT(wparam);
+	UINT isMenuActive = 0;
+
+	WORD charResult = 0;
+
+	int isDeadKey = ToAscii(
```
Comment (MAPJe71) [inline src/Main.cpp]: Why declare a variable and assign the return value when it's not used?

**Review by MAPJe71 — CHANGES_REQUESTED** [review]

File: src/Main.cpp (hunk 2 above)
Comment (visorz) [reply src/Main.cpp]: I intended to use the variable name to document the used function. That function is essential for the new method of character comparison. Although its return value is not used, the reader can understand it fully without looking it up in the Windows API documentation.

**Comment by visorz** [comment]
There is no hint to what caused the continous-integration build to fail. Can someone please restart the build?

**Comment by dail8859** [comment]
> does not use `if .. else ..` statements for all possible combinations. Therefore I am convinced, that my code is easier to read and to maintain

Honestly I was actually a bit confused why `SetPrePostCharacters()` was called consecutively when a normal `if ... else` statement should suffice. 

> There is no hint to what caused the continous-integration build to fail.

Appveyor reported `Cannot fail over job to clouds/groups 'pro-backup'. All clouds within the group are either failing, or offline, or busy` so looks like its not related to the code at all.

Since both this and #5 do nearly the exact same thing there really isn't a way to incorporate both. I do greatly appreciate your time and PR but I'll merge in #5 in favor of this one.

**Pull request closed without merging**